### PR TITLE
PDI-13736 - Regression: PDI Server Startup Time with invalid database…

### PR DESCRIPTION
**Pull request by Aliaksandr-Kastenka**
… connections

### CODE REVIEW

**Comment by Aliaksandr-Kastenka** [comment]
@rmansoor 
Please review.

**Comment by rmansoor** [comment]
@Aliaksandr-Kastenka  I have reviewed and I asked Nick Baker to review it as well. 

First, the timeout you have specified seems low. Where did you get this information from ?
Also, can we not do this properties method for all the database. It seems odd that we are doing some one off thing for mysql

**Comment by Aliaksandr-Kastenka** [comment]
@rmansoor 
As far as i understand the issue (http://jira.pentaho.com/browse/PDI-13736) is about default timeout for MySQL connections.
socketTimeout is MySQL-specific property (http://dev.mysql.com/doc/connector-j/en/connector-j-reference-configuration-properties.html).

**Comment by Aliaksandr-Kastenka** [comment]
@pentaho-nbaker 
Please review.

File: core/src/org/pentaho/platform/engine/services/connection/datasource/dbcp/PooledDatasourceHelper.java [hunk 1]
```diff
@@ -202,8 +203,17 @@ public static PoolingDataSource setupPooledDataSource( IDatabaseConnection datab
...
        */
-      ConnectionFactory factory =
-          new DriverManagerConnectionFactory( url, databaseConnection.getUsername(), databaseConnection.getPassword() );
+      ConnectionFactory factory = null;
+      if( url.contains( "mysql" ) ) {
+        Properties props = new Properties();
+        props.put( "user", databaseConnection.getUsername() );
+        props.put( "password", databaseConnection.getPassword() );
+        props.put( "socketTimeout", "0" );
+        props.put( "connectTimeout", "100" );
```
Comment (pentaho-nbaker) [inline core/src/org/pentaho/platform/engine/services/connection/datasource/dbcp/PooledDatasourceHelper.java]: Hey I'm "okay" with the conditional branch, but this connectTimeout is way too low. Please increase this to something reasonable like 5000.

**Comment by Aliaksandr-Kastenka** [comment]
@pentaho-nbaker 
Pull-request updated.

File: core/src/org/pentaho/platform/engine/services/connection/datasource/dbcp/PooledDatasourceHelper.java [hunk 2]
```diff
@@ -202,8 +203,17 @@ public static PoolingDataSource setupPooledDataSource( IDatabaseConnection datab
        * ConnectionFactory creates connections on behalf of the pool. Here, we use the DriverManagerConnectionFactory
        * because that essentially uses DriverManager as the source of connections.
        */
-      ConnectionFactory factory =
-          new DriverManagerConnectionFactory( url, databaseConnection.getUsername(), databaseConnection.getPassword() );
+      ConnectionFactory factory = null;
+      if( url.contains( "mysql" ) ) {
```
Comment (lucboudreau) [inline core/src/org/pentaho/platform/engine/services/connection/datasource/dbcp/PooledDatasourceHelper.java]: This should be changed to url.startsWith("jdbc:mysql:")

**Comment by Aliaksandr-Kastenka** [comment]
@lucboudreau 
Pull-request updated.

**Pull request merged**